### ISSUE TITLE
AWS Billing visibility

### ISSUE THREAD

**Issue by BenTheElder** [kubernetes/k8s.io#4348]
see previously: https://github.com/kubernetes/k8s.io/issues/4141#issuecomment-1272628985

TLDR:

We should have some idea where we're at on consuming the AWS resources, right now I think not even SIG K8s Infra chairs and/or Steering know this. On GCP we have a public report in SIG K8s Infra with a detailed breakdown so we have some idea where we're at with that.

**Comment by Riaankl** [comment]
> /assign @hh @Riaankl @calebamiles /area billing /committee steering

/unassign @calebamiles 
/assign @BobyMCbobs

**Comment by ameukam** [comment]
> /unassign @calebamiles /assign @BobyMCbobs

Oops. 😅

**Comment by sftim** [comment]
Amazon CloudWatch publishes billing-related metrics in the `AWS/Billing` metrics namespace. We can do things with those and maybe aggregate them with other cost data.

**Comment by upodroid** [comment]
I have worked out to how to do this and I have a functional demo to show at this week's meeting.

TL;DR:

- S3 Metrics:
  - Cloudwatch metrics are a bit meh (mix of daily and minute metrics) :(
  - Gathering some of the metrics has some cost implications. I'm not sure exactly how much.
- For Cost:
  - We need to enable CURs and export it to S3
  - Use Glue to write to Athena
  - Crunch the data in Athena
  - Visualise it in Grafana
  - I wouldn't bother using CloudWatch to read billing metrics at all. It is stale and doesn't have as much info as the CURs

On a sidenote, we should probably start looking at using ArgoCD to deploy k8s manifests instead. Knative uses ArgoCD to deploy Kubernetes manifests, have a look at https://github.com/knative/test-infra/issues/3363 for details. 

Notes:
- https://docs.aws.amazon.com/AmazonS3/latest/userguide/configure-request-metrics-bucket.html
- https://docs.aws.amazon.com/cur/latest/userguide/cur-query-athena.html
- https://docs.aws.amazon.com/cur/latest/userguide/cur-create.html Don't forget the resource IDs
- There is a bug in Grafana which is blocking cross account access. This is needed as Athena lives in the Master Payer account and the S3 metrics live in the registry.k8s.io account https://github.com/grafana/grafana/issues/57664

**Comment by Riaankl** [comment]
Great job @upodroid !!

**Comment by BobyMCbobs** [comment]
I built this sometime a go, hopefully it can be put to some good use
https://github.com/kubernetes/k8s.io/tree/main/infra/aws/aws-costexplorer-export

It currently exports all the Cost Explorer reports from the root level account into a bucket and pushes it over to a BigQuery dataset. Will need to modify it to filter it to just Kubernetes related accounts.

**Comment by Riaankl** [comment]
We currenlty have these:
[GCP Data Studio Report](https://lookerstudio.google.com/u/0/reporting/14UWSuqD5ef9E4LnsCD9uJWTPv8MHOA3e/page/tPVn)
[console.cloud.google.com/billing](https://console.cloud.google.com/billing/018801-93540E-22A20E/reports;chartType=STACKED_BAR;timeRange=YEAR_TO_DATE;grouping=GROUP_BY_SERVICE;projects=k8s-artifacts-prod;credits=SUSTAINED_USAGE_DISCOUNT,SPENDING_BASED_DISCOUNT?mods=logs_tg_staging&organizationId=758905017065&project=k8s-artifacts-prod&pli=1)
[AWS Cost](https://us-east-1.console.aws.amazon.com/cost-management/home?region=us-east-1#/cost-explorer?chartStyle=STACK&costAggregate=unBlendedCost&endDate=2023-02-20&excludeForecasting=false&filter=%5B%7B%22dimension%22:%7B%22id%22:%22RecordTypeV2%22,%22displayValue%22:%22Charge%20type%22%7D,%22operator%22:%22EXCLUDES%22,%22values%22:%5B%7B%22value%22:%22Refund%22,%22displayValue%22:%22Refund%22%7D,%7B%22value%22:%22Credit%22,%22displayValue%22:%22Credit%22%7D%5D%7D%5D&futureRelativeRange=CUSTOM&granularity=Daily&groupBy=%5B%22Service%22%5D&historicalRelativeRange=CUSTOM&isDefault=true&reportName=&showOnlyUncategorized=false&showOnlyUntagged=false&startDate=2023-01-01&usageAggregate=undefined&useNormalizedUnits=false)

Working with Kubecost to consolidate information

**Comment by BenTheElder** [comment]
/lifecycle frozen

as a stop gap we publish high level reports on overall spend to the sig slack weekly and in the biweekly meetings we look at the AWS web console via someone who has access.

**Comment by BenTheElder** [comment]
I don't think this is steering's responsibility at this point. We have ensured that we get billing reports posted regularly via the CNCF but those only get high level spend trends across providers.

we still need a solution for the community to drill down into the AWS spend and we have NO pattern for this , for GCP we have the googlegroups under groups/ linked to IAM which can view the bills and we previously used the data studio report

**Comment by mfahlandt** [comment]
I had a chat with @jeefy about this and we agreed it makes sense to look for a good way to drill down cost across the cloud providers. Specific AWS and Azure, also GCP as the big query based dashboard is off by a certain degree.

I took it to my to-do list.

/Assign
